### ISSUE TITLE
SUPER key stops working for click/drag

### ISSUE THREAD

**Issue by hoodust** [prasanthrangan/hyprdots#536]
Sometimes as soon as I login, sometimes after hours of normal use, the SUPER key will randomly stop allowing me to click-drag windows or right-click-drag to resize windows, but other super+key combo bindings still work. If I reload keybindings.conf it will work exactly ONE additional time and then stop again (repeatable). Logging out and back in fixes it. I can even remap the super key to something besides the Win key and it will work the one additional time and then break again.

**Comment by kRHYME7** [comment]
Have you checked your config for maybe some conflicting  $mainMod variable?

**Comment by hoodust** [comment]
> Have you checked your config for maybe some conflicting $mainMod variable?

Everything seems fine. There's only one `$mainMod =` line, and only the default mouse binds:
```
# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/Resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow
```
...unless there's something else you can think of to check? I've really only added a couple apps and changed some of the default bound keys (like making fullscreen mod+space so I can repurpose mod+F) but when this happens those all work, it's just the mouse  ones impacted.

Speaking of which, the mod+scrollwheel breaks too I just realized. idk if it matters but I have a wireless logitech mx master 3, but dragging in general always works. Super weird. Never happened with a laptop (trackpad) I have hyprdots running on.

**Comment by kRHYME7** [comment]
Strange, I will try to reenact this on my system As I am not always using this feature.

In addition for diagnosing, do you have wev or xev  or any utility that checks what keys you pressed, if so, the problem may occur soon try to run wev / xev to see what your SUPER key returns.

**Comment by hoodust** [comment]
yeah I've been googling to see if this came up in hyprland or wayland in general (not that I've seen so far) and that was a recommendation for a different issue, so I have wev now. Can't get it to happen right now for the life of me, but I'll see if wev gives any insight next time.

**Comment by kRHYME7** [comment]
Also check maybe something is messing your keymaps

**Comment by hoodust** [comment]
> Also check maybe something is messing with your keymap.

Not sure what you mean... I haven't daily driven any linux in forever so maybe I'm missing something stupid. Something other than the keybindings.conf?

> are you following now to the latest commit too?

Also not sure... I've been updating packages when the little pacman in the waybar says to :D or something else I need to do?

**Comment by kRHYME7** [comment]
> If I reload keybindings.conf it will work exactly ONE additional time and then stop again (repeatable)

I missed this part. Maybe there is something changing or conflicting with your keybindings. Could you share your hyprland.conf and the keybindings.conf file? You can utilize https://gist.github.com/ to make this discussion cleaner.



> Also not sure... I've been updating packages when the little pacman in the waybar says to :D or something else I need to do?

OK, I'm assuming that you did not pull the latest changes. To be accurate with our assumptions, maybe you could run git pull to get the latest commit on this repository. However, I'm warning you that this would reset your changes on the configs.

You can read from this [discussion](https://github.com/prasanthrangan/hyprdots/discussions/531) for a slightly more detailed way on how to update your configs.

**Comment by hoodust** [comment]
Thanks, I'll look through that article (don't wanna lose all my configs, although I suppose it's not too many I've changed) and then try to pull. I just installed on this pc a couple days ago, so it's a pretty recent pull.

I've gone through [keybindings.conf](https://gist.github.com/hoodust/57ff5e240314966a120b7e16452a06b5) pretty dang thoroughly but another set of eyes is welcome.

hyprland.conf is bone-stock except for adding my monitor native resolution and adding two exec-once calls for openrgb and xscreensaver (the latter sort of working).

**Comment by hoodust** [comment]
> OK, I'm assuming that you did not pull the latest changes. To be accurate with our assumptions, maybe you could run git pull to get the latest commit on this repository and `Hyprdots/Scripts/install.sh -r` . However, I'm warning you that this would reset your changes on the configs.
> 
> You can read from this [discussion](https://github.com/prasanthrangan/hyprdots/discussions/531) for a slightly more detailed way on how to update your configs.

Well I followed that link and used it to exempt lines 4, 12, and 13 but most of it didn't seem to help and I lost many of my settings except some of my waybar customizations. In particular hyprland.conf and keybindings were wiped out and I thought at least those would be safe. Luckily I had manual backups of the configs. Also it broke some other random things, like prediction in kitty stopped working (fixed this by manually running `restore_zsh.sh`), but wasn't too bad to get everything back.

I'm still having the issue, but it doesn't happen all that often. Last time it happened I had been tweaking OpenRGB settings for a long time, but I don't know if that's related (iirc it happened before I even installed OpenRGB).

Also not sure if it's related, but I had installed Arch with the systemd-boot bootloader prior to installing hyprdots. This might explain some video performance/problems I've had. Neofetch shows both integrated/intel and discrete/nvidia GPUs, but says the driver is my motherboard? I've gotten grub installed now, but just get a black screen after logging in, so I'm guessing there is some microcode that needs to be added. Again, not sure if it's related to the Super key problem, but is there a way to fix that without running the entire install.sh -r script and breaking a bunch of customizations again?

Thanks!

**Comment by hoodust** [comment]
Booting with systemd-boot, neofetch:
```
CPU : 12th Gen Intel i5-12600K (16) @ 4.900GHz 
GPU : NVIDIA GeForce RTX 2080 Ti Rev. A 
GPU : Intel AlderLake-S GT1 
GPU Driver : Micro-Star International Co., Ltd. [MSI] AlderLake-S GT1 [1462:7d30] 
Memory : 2881MiB / 31877MiB 
```
Waybar shows GPU as nvidia though:
![image](https://github.com/prasanthrangan/hyprdots/assets/69404732/d54bc6fb-dc5e-4164-a749-999745a49961)

**Comment by kRHYME7** [comment]
I don't think systemd boot gives this issue as long as you added  Kernel parameters for you gpus. It was same as grub too. Upon installation on this configs I am already using systemd-boot and everything works okay.

**Comment by kRHYME7** [comment]
as for the configs You could get back your configs to the latest cfg directory on your .config folder.

**Comment by hoodust** [comment]
I discovered that I can _sometimes_ recreate this when I use the Fn+Up/Down arrow keys on my keyboard to change its led brightness and then use Super+mouse combos immediately after. This makes zero sense to me because the Super key isn't locked (still works for non-mouse binds) and the Fn key isn't recognized by wev at all, but I hadn't used them or had the issue in almost a week, but twice I was able to trigger it by doing that. 

Unless you have any ideas how that could possibly impact Hypr's mouse binds, we can probably close this one out since it's such a fringe case.
